# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../opensc/gcc4-hotfix.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- opensc-0.9.6/src/pam/pam_support.c.orig	2005-09-09 17:09:49.902472482 +0200
+++ opensc-0.9.6/src/pam/pam_support.c	2005-09-09 17:14:18.045428423 +0200
@@ -205,7 +205,8 @@
 
 static void _cleanup(pam_handle_t * pamh, void *x, int error_status)
 {
-	_pam_delete((char *) x);
+	char *str = (char *) x;  
+	_pam_delete(str);
 }
 
 /* ************************************************************** *
